cli/src/alluxio.org/cli/cmd/exec/class.go

```diff
@@ -0,0 +1,68 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package exec
+
+import (
+	"github.com/spf13/cobra"
+
+	"alluxio.org/cli/env"
+)
+
+var Class = &ClassCommand{
+	BaseJavaCommand: &env.BaseJavaCommand{
+		CommandName:   "class",
+		JavaClassName: "",
+	},
+}
+
+type ClassCommand struct {
+	*env.BaseJavaCommand
+	mainClass string
+	jarFile   string
+	module    string
+}
+
+func (c *ClassCommand) Base() *env.BaseJavaCommand {
+	return c.BaseJavaCommand
+}
+
+func (c *ClassCommand) ToCommand() *cobra.Command {
+	cmd := c.Base().InitRunJavaClassCmd(&cobra.Command{
+		Use:   "class",
+		Short: "Run the main method of an Alluxio class.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return c.Run(args)
+		},
+	})
+	cmd.Flags().StringVar(&c.jarFile, "jar", "",
+		"Determine a JAR file to run.")
+	cmd.Flags().StringVar(&c.module, "m", "",
+		"Determine a module to run.")
+	return cmd
+}
+
+func (c *ClassCommand) Run(args []string) error {
+	var javaArgs []string
+	if c.jarFile != "" {
+		javaArgs = append(javaArgs, "-jar", c.jarFile)
+	} else if c.module != "" {
+		javaArgs = append(javaArgs, "-m", c.module)
+	} else {
+		if len(args) != 0 {
+			c.JavaClassName = args[0]
+		}
+	}
```

Comment:
I agree, this part of code design comes from the need of running code like `bin/alluxio runClass alluxio.stress.cli.StressJobServiceBench --base alluxio://localhost:19998/stress-job-service-base --operation CreateFiles`. To resolve the flags to be sent to the java class (e.g. --operation flag), it needs to put all arguments except the first one (class name) to java arguments. I'm thinking about polishing this logic.

Reply:
`cmd.MarkFlagsMutuallyExclusive(path, fileId)` is an example of having two flags and cannot set both of them.
however, i dont think we have a way to also say "expect exactly one to be set, cannot leave them all empty" because this PR just merged days ago: https://github.com/spf13/cobra/pull/1952

Reply:
also if none of the arguments are specified, it should fail with an error

Reply:
also if none of the arguments are specified, it should fail with an error